pkg/virtualization/core/job/virtualsystemjob.go

```diff
@@ -160,6 +161,13 @@ func (vmjob *VirtualSystemJob) GetException() error {
 		errorCode, _ := vmjob.GetPropertyErrorCode()
 		errorDescription, _ := vmjob.GetPropertyErrorDescription()
 		errorSummaryDescription, _ := vmjob.GetPropertyErrorSummaryDescription()
+
+		if errorCode == 0 {
+			if strings.Contains(errorSummaryDescription, errors.OutOfMemoryErrorSummary) {
```

Comment:
Let's make this Generic. If the errorCode is 0 then, let's create the WMI error from errorSummaryDescription (using NewWMIErrorString as I suggested in another comment)